### PR TITLE
Client manager, Client Grpc Service, clients CRUD operation

### PR DESCRIPTION
**Pull request by erick20**

### CODE REVIEW

**Review by VardanGhazaryan1997 — APPROVED** [review]

File: src/ClientManager/Travely.ClientManager.Repository/Entity/Turist.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+namespace Travely.ClientManager.Repository.Entity
+{
+    [Table("Turist")]
+    public class Turist : BaseEntity
+    {
+        public string FirstName { get; set; }
+        public string LastName { get; set; }
+        public string PhoneNumber { get; set; }
+        public string Email { get; set; }
```
Comment (HAghajanyan-ST) [inline src/ClientManager/Travely.ClientManager.Repository/Entity/Turist.cs:14]: Mark all option fields as nullable (for example `public string? Email { get; set; }`)

File: src/ClientManager/Travely.ClientManager.Repository/Entity/TuristPreference.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations.Schema;
+using System.Text;
+
+namespace Travely.ClientManager.Repository.Entity
+{
+    [Table("Preference")]
+    public class Preference : BaseEntity
```
Comment (HAghajanyan-ST) [inline src/ClientManager/Travely.ClientManager.Repository/Entity/TuristPreference.cs:9]: Rename file to match with class name

File: src/ClientManager/Travely.ClientManager.Repository/Repository/BaseRepository.cs [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+            if (predicate == null)
+            {
+                return set;
+            }
+
+            return set.Where(predicate);
+        }
+
+        public IQueryable<T> Get(params string[] includes)
```
Comment (HAghajanyan-ST) [inline src/ClientManager/Travely.ClientManager.Repository/Repository/BaseRepository.cs:56]: You have code duplication in this 2 methods `IQueryable<T> Get(Expression<Func<T, bool>> predicate, params string[] includes)` and `IQueryable<T> Get(params string[] includes)`

File: src/ClientManager/Travely.ClientManager.Repository/Repository/BaseRepository.cs [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+            {
+                set = set.Include(includes[i]);
+            }
+
+            return set;
+        }
+
+        public IQueryable<T> GetNoTracking(Expression<Func<T, bool>> predicate, params string[] includes)
+        {
+            IQueryable<T> set = _dbContext.Set<T>();
```
Comment (HAghajanyan-ST) [inline src/ClientManager/Travely.ClientManager.Repository/Repository/BaseRepository.cs:70]: Same duplication here

File: src/ClientManager/Travely.ClientManager.Repository/Repository/BaseRepository.cs [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+            {
+                return set;
+            }
+
+            return set.AsNoTracking().Where(predicate);
+        }
+
+        public IQueryable<T> GetNoTracking(params string[] includes)
+        {
+            IQueryable<T> set = _dbContext.Set<T>();
```
Comment (HAghajanyan-ST) [inline src/ClientManager/Travely.ClientManager.Repository/Repository/BaseRepository.cs:87]: Same

File: src/ClientManager/Travely.ClientManager.Repository/Repository/BaseRepository.cs [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+
+        #endregion
+
+        #region SaveChanges
+        public int SaveChanges()
+        {
+            return _dbContext.SaveChanges();
+        }
+
+        public async Task<int> SaveChangesAsync()
```
Comment (HAghajanyan-ST) [inline src/ClientManager/Travely.ClientManager.Repository/Repository/BaseRepository.cs:125]: No need to make method async , you can just return `_dbContext.SaveChangesAsync()` without awaiting it and await in relevant place

File: src/ClientManager/Travely.ClientManager.Repository/Travely.ClientManager.Repository.csproj [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+
+  <PropertyGroup>
+    <TargetFramework>net5.0</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="5.0.2" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore.Relational" Version="5.0.2" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="5.0.2" />
+    <PackageReference Include="Microsoft.Extensions.Configuration.Abstractions" Version="5.0.0" />
```
Comment (HAghajanyan-ST) [inline src/ClientManager/Travely.ClientManager.Repository/Travely.ClientManager.Repository.csproj:11]: No need to reference `Microsoft.Extensions.Configuration.Abstractions` in Repository Lib

File: src/ClientManager/Travely.ClientManager.Service/Travely.ClientManager.Service.csproj [hunk 8]
```diff
@@ -1,15 +1,30 @@
...
+    </PackageReference>
+  </ItemGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\Travely.ClientManager.Repository\Travely.ClientManager.Repository.csproj" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <Protobuf Include="..\..\..\Proto\client.proto" GrpcServices="Server"  Link="Protos\client.proto"/>
+    <Protobuf Include="..\..\..\Proto\preference.proto" GrpcServices="Server"  Link="Protos\preference.proto"/>
```
Comment (HAghajanyan-ST) [inline src/ClientManager/Travely.ClientManager.Service/Travely.ClientManager.Service.csproj:27]: You have 2 protos but using only 1 remove  `preference.proto` or add mapping in `Startup.cs`

File: src/IdentityManager/DataService/Class1.cs [hunk 9]
```diff
@@ -0,0 +1,8 @@
+﻿using System;
+
+namespace IdentityManager.Lib
```
Comment (HAghajanyan-ST) [inline src/IdentityManager/DataService/Class1.cs:3]: Remove unused class

File: src/IdentityManager/WebApi/Controllers/WeatherForecastController.cs [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace IdentityManager.API.Controllers
+{
+    [ApiController]
+    [Route("[controller]")]
+    public class WeatherForecastController : ControllerBase
```
Comment (HAghajanyan-ST) [inline src/IdentityManager/WebApi/Controllers/WeatherForecastController.cs:12]: Remove dummy controller

File: src/IdentityManager/WebApi/WeatherForecast.cs [hunk 11]
```diff
@@ -0,0 +1,15 @@
+using System;
+
+namespace IdentityManager.API
+{
+    public class WeatherForecast
```
Comment (HAghajanyan-ST) [inline src/IdentityManager/WebApi/WeatherForecast.cs:5]: Remove

**Review by HAghajanyan-ST — CHANGES_REQUESTED** [review]

File: src/ClientManager/Travely.ClientManager.Service/Travely.ClientManager.Service.csproj (hunk 8 above)
Comment (erick20) [reply src/ClientManager/Travely.ClientManager.Service/Travely.ClientManager.Service.csproj:27]: the preference proto not implemented yet, we planned to do that in another branch

File: src/IdentityManager/DataService/Class1.cs (hunk 9 above)
Comment (erick20) [reply src/IdentityManager/DataService/Class1.cs:3]: this is not our part, so as we merged to our branch from master, this class came from identity, and when they fix this we can merge again

File: src/IdentityManager/WebApi/Controllers/WeatherForecastController.cs (hunk 10 above)
Comment (erick20) [reply src/IdentityManager/WebApi/Controllers/WeatherForecastController.cs:12]: this is not our part, so as we merged to our branch from master, this class came from identity, and when they fix this we can merge again

File: src/IdentityManager/WebApi/WeatherForecast.cs (hunk 11 above)
Comment (erick20) [reply src/IdentityManager/WebApi/WeatherForecast.cs:5]: this one came from identity too

File: src/ClientManager/Travely.ClientManager.Repository/Entity/BaseEntity.cs [hunk 12]
```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+using System.Text;
+
+namespace Travely.ClientManager.Repository.Entity
+{
+    public class BaseEntity
```
[the remark this answers is not in the document]
Comment (erick20) [reply src/ClientManager/Travely.ClientManager.Repository/Entity/BaseEntity.cs:9]: Hov jan in our first version we had the Abstraction class library, and all entities and interfaces were there, but after the discussion with Arthur, he requested to move all the the classes and interfaces that connected with Data Access layer(in our case repository) into Repository class Lib.

File: src/ClientManager/Travely.ClientManager.Repository/Abstraction/ITuristRepository.cs [hunk 13]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Travely.ClientManager.Repository.Entity;
+
+namespace Travely.ClientManager.Repository.Abstraction
+{
+    public interface ITuristRepository : IBaseRepository<Turist>
```
[the remark this answers is not in the document]
Comment (erick20) [reply src/ClientManager/Travely.ClientManager.Repository/Abstraction/ITuristRepository.cs:8]: will fix   :)))

**Review by HAghajanyan-ST — APPROVED** [review]

**Review by VardanGhazaryan1997 — APPROVED** [review]

**Pull request merged**